specreduce/background.py

```diff
@@ -369,15 +375,17 @@ def sub_image(self, image=None):
 
         Returns
         -------
-        `~specutils.Spectrum1D` object with same shape as ``image``.
+        spec : `~specutils.Spectrum1D`
+            Spectrum object with same shape as ``image``.
         """
         image = self._parse_image(image)
 
-        # a compare_wcs argument is needed for Spectrum1D.subtract() in order to
+        # a compare_wcs argument is needed for Spectrum.subtract() in order to
         # avoid a TypeError from SpectralCoord when image's spectral axis is in
         # pixels. it is not needed when image's spectral axis has physical units
         kwargs = {"compare_wcs": None} if image.spectral_axis.unit == u.pix else {}
-
+        if not SPECUTILS_LT_2:
+            kwargs["spectral_axis_index"] = image.flux.ndim - 1
         # https://docs.astropy.org/en/stable/nddata/mixins/ndarithmetic.html
         return image.subtract(self.bkg_image(image), **kwargs)
```

Comment:
```suggestion
        return image - self.bkg_image(image)
```
These two suggestions depend on https://github.com/astropy/specutils/pull/1222, combined with that I think this PR should be good to handle specutils 2.0.